crates/chain-state/src/in_memory.rs

```diff
@@ -268,6 +279,25 @@ impl CanonicalInMemoryState {
         }
     }
 
+    /// Performs a range lookup for block headers in the in-memory state and returns partial or full
+    /// results.
+    pub fn range_lookup(&self, start: u64, end: u64) -> RangeLookupResult<Header> {
```

Comment:
I think we can just accept a range here, it makes it less ambigious whether end is inclusive or not since it is up to the user

Reply:
We would still have to pass self.canonical_in_memory_state.get_canonical_block_number() through range_lookup() and also import convert_range_bounds() to in_memory.rs, right?  It seems a little messy, is there a better way to have range_lookup() accept range?